code/game/objects/structures/spawner.dm

```diff
@@ -14,6 +14,44 @@
 	var/spawn_text = "emerges from"
 	var/faction = list(FACTION_HOSTILE)
 	var/spawner_type = /datum/component/spawner
+	/// Is this spawner taggable with a mining scanner?
+	var/scanner_taggable = FALSE
+	/// Has this spawner been tagged/analyzed by a mining scanner?
+	var/gps_tagged = FALSE
+	/// A short identifier for the mob it spawns. Keep around 3 characters or less?
+	var/mob_gps_id = "???"
+	/// A short identifier for what kind of spawner it is, for use in putting together its GPS tag.
+	var/spawner_gps_id = "Creature Nest"
+	/// A complete identifier. Generated on tag (if tagged), used for its examine.
+	var/assigned_tag
+
+/obj/structure/spawner/examine(mob/user)
+	. = ..()
+	if(scanner_taggable)
+		if(!gps_tagged)
+			. += span_notice("It looks like you could probably scan and tag it with a <b>mining analyzer</b>.")
+		else
+			. += span_notice("A holotag's been attached, projecting \"<b>[assigned_tag]</b>\".")
+
+/obj/structure/spawner/attackby(obj/item/item, mob/user, params)
+	if(scanner_taggable)
+		if(!istype(item, /obj/item/mining_scanner) && !istype(item, /obj/item/t_scanner/adv_mining_scanner))
+			return ..()
+		gps_tag(user)
```

Comment:
```suggestion
	var/static/list/scanner_types = list( /obj/item/mining_scanner, /obj/item/t_scanner/adv_mining_scanner)
	if(scanner_taggable && is_type_in_list(item, scanner_types)
		gps_tag(user)
```
Alternately define `scanner_types` on the typepath so that you can have something markable by some other item.